server/backend/api/app/crud/crud_user_terms_conditions.py

```diff
@@ -0,0 +1,64 @@
+import logging
+from sqlalchemy.orm import Session
+from http import HTTPStatus
+
+from api.app.models.model import FamUserTermsConditions
+from api.app.schemas import Requester
+from api.app.utils.utils import raise_http_exception
+from api.app.crud.crud_utils import is_requester_external_delegated_admin
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_user_terms_conditions_by_user_id_and_version(
+    db: Session, user_id: int, version: str
+):
+    return (
+        db.query(FamUserTermsConditions)
+        .filter(
+            FamUserTermsConditions.user_id == user_id,
+            FamUserTermsConditions.version == version,
+        )
+        .one_or_none()
+    )
+
+
+def require_accept_terms_and_conditions(
+    db: Session, requester: Requester, version: str
+) -> bool:
+    """
+    Return False if found record (means user already accepted terms and conditions)
+    Return True if not found (means user needs to accept terms and conditions)
+    """
+    if is_requester_external_delegated_admin(
+        db, requester
+    ) and not get_user_terms_conditions_by_user_id_and_version(
+        db, requester.user_id, version
+    ):
+        return True
+    return False
+
+
+def create_user_terms_conditions(
+    db: Session, user_id: int, version: str, requester: str
+) -> FamUserTermsConditions:
+    LOGGER.debug(
+        f"Creating user terms conditions acceptance record for user {user_id} and version {version}"
+    )
+
+    if get_user_terms_conditions_by_user_id_and_version(db, user_id, version):
+        error_msg = "User already accepted terms and conditions."
+        raise_http_exception(status_code=HTTPStatus.CONFLICT, error_msg=error_msg)
```

Comment:
Hi Ian, SonarCloud is complaining the type of the status code, maybe the type of the status_code in the method should be a number instead of HTTPStatus? 
<img width="935" alt="Screen Shot 2024-06-13 at 11 07 31 AM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/b68ec0b4-1c89-465c-b95b-fea4632d6ff8">
<img width="1296" alt="image" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/08ae2faf-00c8-49b3-a9bc-1c63426601b0">

Reply:
As discussed, although I looked into it but don't find how it can be incorrect type. I am confused with the Sonar report as well. The one thing I am thinking of is maybe to do an experiment on adding third (default) argument in (with the order of the function) and see if that Sonar issue disappear... but I am good to ignore it if you like...